### PR TITLE
[coverage] Speed up all tests x2

### PR DESCRIPTION
**Pull request by gpotter2**
This PR:
- brings a new feature: multi file support in UTscapy.
That allow us to use the same scapy instance to run multiple tests, and speed up the tests. (also added a new log level with more user-friendly texts)
- brings a new feature: JSON config files in UTscapy. Optional, it reduces the command line complexity and allow to use complex `PREEXEC` structures
- Fixes some bugs with the tests (`_` used with `xrange` made `_` a static value, removed `sys.__stdout__` usage)
- Fixes `fixname()` function not being called in DaDict

(PR won't work without the bug fixes)

Results:
- Travis CI:
from an [older build](https://travis-ci.org/gpotter2/scapy/builds/207658839), built in 9 min 19 sec, we're [now](https://travis-ci.org/gpotter2/scapy/builds/207800650) at 4 min 48 sec
- Appveyor: even better,
from [older](https://ci.appveyor.com/project/gpotter2/scapy/build/1.0.296) in 9 min 21 sec we're [now](https://ci.appveyor.com/project/gpotter2/scapy/build/1.0.298) at 3 min 56 sec.

So in average, that should speed up the tests (running part only), by 2.

### CODE REVIEW

**Comment by gpotter2** [comment]
@p-l- or @guedou, if you could review that too it would be great :)

File: scapy/dadict.py [hunk 1]
```diff
@@ -25,7 +25,7 @@ class DADict_Exception(Scapy_Exception):
 class DADict:
     def __init__(self, _name="DADict", **kargs):
         self._name=_name
-        self.__dict__.update(kargs)
+        self._update(kargs)
```
Comment (p-l-) [inline scapy/dadict.py]: What is the purpose of this change?

File: test/regression.uts [hunk 2]
```diff
@@ -1217,7 +1217,7 @@ in6_addrtomac("FE80::" + in6_mactoifaceid("FF:00:00:00:00:00", ulbit=0)) == 'ff:
 import socket
 
 res=True
-for _ in xrange(10):
+for a in xrange(10):
```
Comment (p-l-) [inline test/regression.uts]: What is the purpose of this change?

File: scapy/dadict.py (hunk 1 above)
Comment (gpotter2) [inline scapy/dadict.py]: That's a fix for a bug that would be trigger by this PR:
As you can see, `__setitem__` uses the `fixname` function before setting the items. But for a reason, using `self.__dict__.update` do not use `__setitem__`, so what was provided in `__init__` was not fixed.

File: test/regression.uts (hunk 2 above)
Comment (gpotter2) [inline test/regression.uts]: That's a really hidden bug.
Using this replaces `_` by the `xrange` iterator, so once the for is finished, `_` will be stuck with the value `9` for ever.

Comment (p-l-) [inline test/regression.uts]: ok and you need `_` after that I get it. Normally that's not a problem but `_` is actually used in the tests!

File: scapy/dadict.py (hunk 1 above)
Comment (p-l-) [inline scapy/dadict.py]: All right! Maybe we could just overwrite `.update` in that case? Would it work?

File: test/regression.uts (hunk 2 above)
Comment (gpotter2) [inline test/regression.uts]: Exactly

File: scapy/dadict.py (hunk 1 above)
Comment (p-l-) [inline scapy/dadict.py]: Actually I even think it would make sense to just name your method `.update()`.

Comment (gpotter2) [inline scapy/dadict.py]: Renaming it `update` does not work (it sill call the __dict__ one).
__dict__.update is a readonly value, so impossible to overwrite...

Comment (gpotter2) [inline scapy/dadict.py]: Renamed !

**Comment by p-l-** [comment]
Nice @gpotter2. Just curious: what was the problem with renaming `.update()`?

**Comment by gpotter2** [comment]
I thought maybe simply by renaming it `.update()`, calling `self.__dict__.update` would have used it... That wasn't the case
